src/types/refs.rs

```diff
@@ -138,6 +149,15 @@ unsafe impl<T: ScriptClass> Sync for Ref<T> {}
 pub struct WeakRef<T: ScriptClass>(BaseRef<NativeType<T>>);
 
 impl<T: ScriptClass> WeakRef<T> {
+    #[inline]
+    pub(super) fn from_raw(
+        value: &red::SharedPtrBase<<<T as ScriptClass>::Kind as ClassKind<T>>::NativeType>,
```

Comment:
```suggestion
        value: BaseRef<NativeType<T>>,
```
I think we should expect our own BaseRef here

Reply:
casting `WeakHandle<_>` to `WeakRef<_>` allows to completely get rid of this method.